### PR TITLE
chore: add docker info in readme & release, update docker task to include partial versions

### PR DESCRIPTION
**Pull request by Animeshz**
Remaining task of #10 (let me know if any changes are needed).

### CODE REVIEW

File: hack/push-docker-image.sh [hunk 1]
```diff
@@ -25,14 +27,17 @@ done
...
-        "${IMAGE}:${TAG}-amd64" \
-        "${IMAGE}:${TAG}-arm64" \
-        "${IMAGE}:${TAG}-arm"
 
-# Annotate to set which image is build for which CPU architecture
-for arch in $CPU_ARCHS; do
-  docker manifest annotate --arch "$arch" "${IMAGE}:${TAG}" "${IMAGE}:${TAG}-$arch"
-done
-docker manifest push "${IMAGE}:${TAG}"
+for EXP_TAG in ${TAGS[@]}; do
```
Comment (orphaner) [inline hack/push-docker-image.sh]: OK, this is great to iterate here

File: hack/push-docker-image.sh [hunk 2]
```diff
@@ -15,8 +15,10 @@ set -e
 # only exit with zero if all commands of the pipeline exit successfully
 set -o pipefail
 
-CPU_ARCHS="amd64 arm64 arm"
+echo ${TAG} | IFS="." read major minor revision
```
Comment (orphaner) [inline hack/push-docker-image.sh]: this breaks the usage of this script in the snapshot workflow. 
I'm wondering if there is a way to call this script like: `TAGS="1.0.0 1.0 1" push-docker-image.sh` and exclude the logic of calculating the tags from here to the release workflow. Is there a github action to calculate the tags?

**Review by orphaner — CHANGES_REQUESTED** [review]

File: hack/push-docker-image.sh (hunk 2 above)
Comment (Animeshz) [reply hack/push-docker-image.sh]: @orphaner Yes, you could not change `IFS`, which defaults to space character ` `. But where would you send this information from (like in workflow)?

I'm also not sure how snapshot would look like, if snapshot is to look something like vx.y.z-rc1 or -alpha1 etc then it'll be better not to override the `x.y.z`, `x.y` and `x.` versions, they should be overridden in stable release isn't it? Maybe just check if -rc{x} exist in the snapshot and if it does do nothing, or else do as it is doing?

**Comment by orphaner** [comment]
hello @Animeshz , sorry for the delay on that. I had a hard past week at work and this week-end I've played building lego with my little boy :)

regarding this pull request, let me clarify. In the `release.yml` workflow there is a missing step to build the operator docker image. 
In this new step we could imagine using [this action](https://github.com/jupyterhub/action-major-minor-tag-calculator) to get the desired tag list.
The `push-docker-image.sh` should still take the `TAG` env var ; that may be a list separated by a ,

I've created a slack channel so that we could discuss more easily: https://kidle-dev.slack.com/archives/C02JXP2JTK2

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -133,6 +133,23 @@ jobs:
...
+
+      - name: Retrieve Docker Image Tags
+        uses: jupyterhub/action-major-minor-tag-calculator@v1
+        id: dockertags
+        with:
+          githubToken: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and Push Multi-arch Docker Image
+        run: |
+          TAG=${{env.GITHUB_REF_SLUG}} TAGS=${{ join(fromJson(steps.dockertags.outputs.tags)) }} make WHAT=operator build-multi-arch-image
```
Comment (orphaner) [inline .github/workflows/release.yml:151]: `TAG` is not necessary anymore ; `TAGS` can handle a single value

File: .github/workflows/snapshot.yaml [hunk 4]
```diff
@@ -56,13 +56,19 @@ jobs:
...
+        githubToken: ${{ secrets.GITHUB_TOKEN }}
+
     - name: "fmt & vet"
       run: |
         make WHAT=operator fmt vet
 
     - name: "Build and Push multi arch Operator Image"
       run: |
-        TAG=${{env.GITHUB_REF_SLUG}} make WHAT=operator build-multi-arch-image
+        TAG=${{env.GITHUB_REF_SLUG}} TAGS=${{ join(fromJson(steps.dockertags.outputs.tags)) }} make WHAT=operator build-multi-arch-image
```
Comment (orphaner) [inline .github/workflows/snapshot.yaml:71]: `TAGS=${{env.GITHUB_REF_SLUG}} make WHAT=operator build-multi-arch-image`
should be OK

File: hack/push-docker-image.sh [hunk 5]
```diff
@@ -17,22 +17,25 @@ set -o pipefail
...
-        "${IMAGE}:${TAG}-arm64" \
-        "${IMAGE}:${TAG}-arm"
 
-# Annotate to set which image is build for which CPU architecture
-for arch in $CPU_ARCHS; do
-  docker manifest annotate --arch "$arch" "${IMAGE}:${TAG}" "${IMAGE}:${TAG}-$arch"
-done
-docker manifest push "${IMAGE}:${TAG}"
+IFS=','
+for EXP_TAG in ${TAGS}; do
```
Comment (orphaner) [inline hack/push-docker-image.sh:29]: now there is just an iteration over `TAGS` ; 
`EXP_TAG` has to renamed to `TAG`

File: hack/push-docker-image.sh [hunk 6]
```diff
@@ -17,22 +17,25 @@ set -o pipefail
...
+  # Create manifest to join all images under one virtual tag
+  docker manifest create -a "${IMAGE}:${EXP_TAG}" \
```
Comment (orphaner) [inline hack/push-docker-image.sh:31]: `EXP_TAG` => `TAG`

File: hack/push-docker-image.sh [hunk 7]
```diff
@@ -17,22 +17,25 @@ set -o pipefail
...
+          "${IMAGE}:${TAG}-amd64" \
+          "${IMAGE}:${TAG}-arm64" \
+          "${IMAGE}:${TAG}-arm"
+
+  # Annotate to set which image is build for which CPU architecture
+  for arch in $CPU_ARCHS; do
+    docker manifest annotate --arch "$arch" "${IMAGE}:${EXP_TAG}" "${IMAGE}:${TAG}-$arch"
```
Comment (orphaner) [inline hack/push-docker-image.sh:38]: `EXP_TAG` => `TAG`

File: hack/push-docker-image.sh [hunk 8]
```diff
@@ -17,22 +17,25 @@ set -o pipefail
...
+  done
+  docker manifest push "${IMAGE}:${EXP_TAG}"
```
Comment (orphaner) [inline hack/push-docker-image.sh:40]: `EXP_TAG` => `TAG`

**Review by orphaner — CHANGES_REQUESTED** [review]
Hello; we are almost good for this PR, congrats :tada:

File: .github/workflows/release.yml (hunk 3 above)
Comment (Animeshz) [reply .github/workflows/release.yml:151]: @orphaner TAG is used by line number 22:

```bash
make WHAT=$WHAT docker GOARCH="$arch"
```

Comment (Animeshz) [reply .github/workflows/release.yml:151]: Its build once, tag many.

Comment (orphaner) [reply .github/workflows/release.yml:151]: ho yes, my bad

**Comment by orphaner** [comment]
@Animeshz I've just accepted your PR; thanks again for your help :+1:

**Pull request merged**